src/audio/dsm/sof_dsm_inf.c

```diff
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Maxim Integrated All rights reserved.
+//
+// Author: Ryan Lee <ryans.lee@maximintegrated.com>
+
+#include <sof/audio/sof_dsm.h>
+#include <sof/audio/component.h>
+#include <sof/trace/trace.h>
+#include <sof/drivers/ipc.h>
+#include <sof/ut.h>
+
+#include <sof/audio/channel_map.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <sof/bit.h>
+#include <sof/common.h>
+#include <ipc/channel_map.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+static void *_wrap_buf_ptr(void *ptr, const struct audio_stream *buffer)
```

Comment:
Dont we already have this as an external API ? If so we should use the generic version.

Reply:
You are right. Removed this function and reused generic functions.